### ISSUE TITLE
Our screen center is not quite centered

### ISSUE THREAD

**Issue by sonora** [osmandapp/OsmAnd#10108]
It is a minor issue but has bugged me for some time in one-handed operation:

Make sure you are in a profile where the screen center is supposed to be in the middle, not near the bottom of the screen. Now tap the blue MyPosition button so that your position should be shown exactly in the screen center.

But in fact it is not, it is shown slightliy below it. You will only notice if you use our one finger zoom in, i.ie. zoom in by double tapping with your thumb but do not lift the thumb after the second tap, instead slide it downwards. You will observe that the position marker on the screen will drift downwards instead of staying centeted, because it initially did not coincide exactly withe the geometric screen center (by maybe the width of the Android navigation bar or so).

It could be device dependent, not sure, but I think we simply have a small inconsistency in our code when calculating the screen center for centering vs. for zooming operations.

**Comment by vshcherb** [comment]
It's because status bar is considered to be part of the screen (sometimes is transparent and it's taken into account).

**Comment by sonora** [comment]
Yes, exactly, and that's no problem, as long as we do it consistently, (i.e. both for the zoom center and where we center when someone taps the "center my location" button). But it seems that in one case we include the width of the status bar, in the other case not, and that creates the issue, right?

**Comment by vshcherb** [comment]
I don't see the issue it looks consistent on my phone, can you please share a screenshot?

**Comment by sonora** [comment]
Here's a screenshot, you can see how the blue position icon moved down from the zoom-screen center (black double circle).
![Screenshot_20201105-222358_OsmAnd~](https://user-images.githubusercontent.com/845735/98298356-29291800-1fb6-11eb-85d1-7b355f9edaca.jpg)

**Comment by vshcherb** [comment]
This is a bug! But it could be a feature if you changed setting to display location at the bottom, when map rotates to direction as in car profile. So it would be good to understand how did you get in that situation

**Comment by sonora** [comment]
I have investigated some more: The issue is **not present** on devices where the status bar (in the screenshot above where the clock time, WiFi icon etc. is shown) is completely **opaque**. There the map screen area taken into account seems only between _the bottom of the status bar and the bottom of the device screen_. (Center the map and use a ruler to verify that MyPosition is displayed exactly half way between the bottom of the status bar and the bottom of the device screen.)

Now take a device where the status bar is shown **semi-transparent** (i.e. with the map shining through). There the MyPosition will be centered _half way between the **top** of the status bar (=top of the device screen) and the bottom of the device screen_.

On those "semi-transparent devices", just do the following:
 - Select a profile where the screen center is displayed in the middle (not at the bottom) of the screen, go to the map screen, zoom out a bit to be sure.
 - After a stable location fix, tap the blue "My Position" button so the screen centers to your position
 - Now perform the '`double-tap + zoom in (via slide down)`' gesture. With increasing zoom-in, the blue MyPosition will separate from the screen center and drift down.

Not a big effect, but annoying for people who use the feature a lot for one-handed operation.
